### ISSUE TITLE
Take action of the warning in latest builds - "patterns aren't allowed in methods without bodies"

### ISSUE THREAD

**Issue by benashford** [benashford/rs-es#93]
A future version of Rust will make an error of this warning, so we should take action sooner rather than later.

```
warning: patterns aren't allowed in methods without bodies, #[warn(patterns_in_fns_without_body)] on by default
  --> src/lib.rs.in:41:25
   |
41 |     fn read_response<R>(mut self) -> Result<R, EsError> where R: Deserialize;
   |                         ^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #35203 <https://github.com/rust-lang/rust/issues/35203>
```

**Comment by benashford** [comment]
This is fixed as part of: https://github.com/benashford/rs-es/pull/95

**Comment by benashford** [comment]
Fixed on `v0.7.1`
